docker/test/installer/linux.Dockerfile

```diff
@@ -6,56 +6,71 @@
 # the terms of the Apache License 2.0 which accompanies this distribution.     #
 # ============================================================================ #
 
-ARG base_image=amd64/almalinux:8
-ARG base_image_mpibuild=${base_image}
+ARG base_image=redhat/ubi8:8.0
+ARG base_image_mpibuild=amd64/almalinux:8
 
 # [OpenMPI Installation]
 FROM ${base_image_mpibuild} as mpibuild
+ARG base_image_mpibuild
 SHELL ["/bin/bash", "-c"]
 ARG DEBIAN_FRONTEND=noninteractive
-RUN dnf install -y --nobest --setopt=install_weak_deps=False \
-        'dnf-command(config-manager)'
 
 ## [Prerequisites]
-ADD scripts/configure_build.sh /cuda-quantum/scripts/configure_build.sh
-RUN source /cuda-quantum/scripts/configure_build.sh install-cuda
-RUN source /cuda-quantum/scripts/configure_build.sh install-gcc
+ADD docker/test/installer/dependencies.sh /runtime_dependencies.sh
+RUN bash runtime_dependencies.sh ${base_image_mpibuild}
 RUN dnf install -y --nobest --setopt=install_weak_deps=False \
-        autoconf libtool flex make wget
+        autoconf libtool flex make wget \
+        gcc-toolset-11 cuda-cudart-devel-11-8
 
 ## [Build]
+ADD scripts/configure_build.sh /cuda-quantum/scripts/configure_build.sh
 RUN source /cuda-quantum/scripts/configure_build.sh build-openmpi
 
 # [CUDA Quantum Installation]
 FROM ${base_image}
 ARG base_image
-
 ARG DEBIAN_FRONTEND=noninteractive
+ARG libstdcpp_package
+ARG cudart_version
+ARG cuda_distribution
+
+## [Runtime dependencies]
 ADD docker/test/installer/dependencies.sh /runtime_dependencies.sh
-RUN bash runtime_dependencies.sh ${base_image}
+RUN export LIBSTDCPP_PACKAGE=${libstdcpp_package} && \
+    export CUDART_VERSION=${cudart_version} && \
+    export CUDA_DISTRIBUTION=${cuda_distribution} && \
+    . /runtime_dependencies.sh ${base_image} && \
+    # working around the fact that the installation of the dependecies includes
+    # setting some environment variables that are expected to be persistent on
+    # on the host system but would not persistent across docker commands
+    env | egrep -v "^(HOME=|USER=|MAIL=|LC_ALL=|LS_COLORS=|LANG=|HOSTNAME=|PWD=|TERM=|SHLVL=|LANGUAGE=|_=)" \
+        >> /etc/environment
 
 ## [MPI Installation]
 COPY --from=mpibuild /usr/local/openmpi/ /usr/local/openmpi
 RUN ln -s /usr/local/openmpi/bin/mpiexec /bin/mpiexec
 
 # Create new user `cudaq` with admin rights to confirm installation steps.
-RUN adduser --disabled-password --gecos '' cudaq && adduser cudaq sudo \
-    && echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
-RUN chown -R cudaq /home/cudaq && chgrp -R cudaq /home/cudaq
+RUN useradd cudaq && mkdir -p /etc/sudoers.d && \
+    echo 'cudaq ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers.d/020_cudaq
```

Comment:
Where did the magic number `020` come from?

Reply:
arbitrarily made up. the number would determine the order in which the script is executed.